### ISSUE TITLE
jsobfu 0.3.0 is noticeably slower

### ISSUE THREAD

**Issue by jvazquez-r7** [rapid7/jsobfu#11]
@wchen-r7 noticed which exploit/multi/browser/firefox_tostring_console_injection loads noticeably slower after trying to update metasploit-framework to use jsobfu 0.3.0. Definitely we should figure out what is going on before updating the used version in the framework.

See https://github.com/rapid7/metasploit-framework/pull/4805 for details.

ping @joevennix because he is the jsobfu master :) but nothing like assignment or nothing like that, I can work on it indeed, I just need to finish some tasks before!

**Comment by joevennix** [comment]
That's very strange. If anything 0.3.0 should be faster. There is not very much changed https://github.com/rapid7/jsobfu/pull/9/files#diff-e048b88a0b505db0d64daa732d120f60L103

The travis jobs have not seen a slowdown, and they are running obfuscation on 700kb js files.

I have a second hash `#merge!` call now, perhaps that is slowing things down?

**Comment by wchen-r7** [comment]
IIRC it doesn't have to be firefox_tostring_console_injection, so it's pretty bizarre. If you want I can double check on the slowdown and provide a more specific report about where I experience it.

**Comment by joevennix** [comment]
@wchen-r7 that would be very helpful indeed :) Even just knowing if its the exploit obfuscation, or the obfuscation of `os.js` might give some clues. I reviewed the code, it seems the only difference performance-wise might be these new `#merge` calls, but these should only be run once per iteration:

    https://github.com/rapid7/jsobfu/commit/f76a0520aed8da0f1e0d3e437025daafb6b63c53#diff-e048b88a0b505db0d64daa732d120f60R84

If it's not that then I'm stumped here.

**Comment by wchen-r7** [comment]
@joevennix So I tried this morning, it was really weird.

At first I tested this way:

```
use exploit/multi/browser/firefox_tostring_console_injection
set jsobfuscate 1
run
```

It took quite awhile to finally start the listener.

And then I did this:

```
jobs -K
use exploit/windows/browser/ms14_064_ole_code_execution
run
```

And then that was slow.

And then I did this:

```
set jsobfuscate 0
run
```

And then that was fast again.

And then I exited msfconsole, did this:

```
use exploit/multi/browser/firefox_tostring_console_injection
run
```

And then it was fast again.

And then I couldn't reproduce the lag anymore. I tried to set jsbofuscate back to 1 again, and all that. The problem is spooky.
